src/NLopt.jl

```diff
@@ -461,9 +461,9 @@ function nlopt_callback_wrapper(
 )::Cdouble
     d = unsafe_pointer_to_objref(d_)::Callback_Data
     x = unsafe_wrap(Array, p_x, (n,))
-    grad = unsafe_wrap(Array, p_grad, (n,))
+    grad = p_grad == C_NULL ? Cdouble[] : unsafe_wrap(Array, p_grad, (n,))
```

Comment:
Should we declare a `const empty_grad = Cdouble[]` and just re-use it, since the caller shouldn't modify it?